lib/virt_autotest/common.pm

```diff
@@ -385,6 +388,9 @@ if (get_var("REGRESSION", '') =~ /xen/) {
         sles15sp4 => {
             vm_name => 'sles-15.4_openQA-virtualization-maintenance',
         },
+        sles15sp4teradata => {
+            vm_name => 'sles-15.4_teradata_openQA-virtualization-maintenance',
```

Comment:
Shall below logic be added to filter the guest?

```delete($guests{sles15sp4teradata}) if (!is_sle('=15-SP4-teradata'));```

Reply:
When it's the TD MU, the TD repo will be added to VM by guest autoyast.  The VM is TD VM. 
we don't need additional VM for TD, right?

Reply:
A little bit concerned about vm_name here. Better not to have underscore and dot ?

Reply:
@alice-suse  sles15sp4teradata guest will  have the name sles15sp4 when running TD tests, no need to add the filter.

@waynechen55 no parsing of the strings is done for hyperV tests, you want it to be like `sles-15.4.teradata_openQA-virtualization-maintenance` this?

Reply:
@tbaev  It seems the point does not reach you well. That's true for kvm and xen cases. However here, you add the new teradata guests for vmware and hyperv, right? My comment for filtering was for somewhere around https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/951c2377c6fbc370f8b78149accc047ced81b6c8/lib/virt_autotest/common.pm#L366, which will modify the %guests hash to only include the guests that you want to test.

Reply:
I have asked Roy for advice too, I am starting to think that the filters for hyperV/VMware are depreciated and not needed anymore. I will make a small new commit if needed.

Reply:
VR with latest chages:

kvm/xen
http://openqa.qam.suse.cz/tests/59811#dependencies
http://openqa.qam.suse.cz/tests/59822#dependencies

migration kvm/xen
http://openqa.qam.suse.cz/tests/59834#dependencies
http://openqa.qam.suse.cz/tests/59837#dependencies

Reply:
> @waynechen55 no parsing of the strings is done for hyperV tests, you want it to be like `sles-15.4.teradata_openQA-virtualization-maintenance` this?

If my memory is correct, there will be trouble with DNS mapping/resolving if vm name has underscore. So I think if there is no mandatory requirements, it would be better do not use underscore and dot to prevent potential issues. Please refer to https://domainkeys.sourceforge.net/underscore.html

Reply:
`15.4.teradata_openQA-virtualization-maintenance` is not the DNS name of the VM, it is the virtual machine name in the hypervisor

Reply:
@tbaev  Hi Teodor, filter looks good to me. Regarding verification runs in https://openqa.suse.de/tests/overview?distri=sle&version=15-SP4-teradata&build=tbaev%3Ateradata&groupid=322, no guest is tested. It seems TERADATA=15-SP4 is not given when triggering job. Can you add a new verification? 

Besides, if my understanding is correct, the OSD job group yaml can delete below with the new PR?
```
  sle-15-SP4-teradata-Server-DVD-Incidents-Virt-x86_64:
    distri: sle
    flavor: Server-DVD-Incidents-Virt
    version: 15-SP4-teradata

...

    sle-15-SP4-teradata-Server-DVD-Incidents-Virt-x86_64:
    - mau-extratests-virt-vmware
    - mau-extratests-virt-hyperv
```